fe/fe-core/src/main/java/com/starrocks/external/CachingRemoteFileIO.java

```diff
@@ -0,0 +1,76 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.external;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.starrocks.connector.exception.StarRocksConnectorException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import static com.google.common.base.Throwables.throwIfInstanceOf;
+import static com.google.common.cache.CacheLoader.asyncReloading;
+import static com.google.common.util.concurrent.MoreExecutors.newDirectExecutorService;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class CachingRemoteFileIO implements RemoteFileIO {
+    public static final long NEVER_EVICT = -1;
+    public static final long NEVER_REFRESH = -1;
+    private final RemoteFileIO fileIO;
+    private final LoadingCache<RemotePathKey, List<RemoteFileDesc>> cache;
+
+    public CachingRemoteFileIO(RemoteFileIO fileIO,
+                               Executor executor,
+                               long expireAfterWriteSec,
+                               long refreshIntervalSec,
+                               long maxSize) {
+        this.fileIO = fileIO;
+        this.cache = newCacheBuilder(expireAfterWriteSec, refreshIntervalSec, maxSize)
+                .build(asyncReloading(CacheLoader.from(this::loadRemoteFiles), executor));
+    }
+
+    public static CachingRemoteFileIO reuseRemoteIO(RemoteFileIO fileIO, long maxSize) {
+        return new CachingRemoteFileIO(
+                fileIO,
+                newDirectExecutorService(),
+                NEVER_EVICT,
```

Comment:
Why Never evict and never refresh?

Reply:
This `reuseRemote` method is used to create a query level instance based on catalog level instance. Its life cycle is only the current query, so no params of cache expire or refresh is required.

Reply:
I think we can offer a interface which can expire cache manually when query finished.

Reply:
Do you mean we need to validate the cache after the query？